src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerHelper.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Org.OData.Core.V1;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Helper class for OData Serialization
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1053:StaticHolderTypesShouldNotHaveConstructors")]
+    public class ODataSerializerHelper
+    {
+        internal static void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext, object value, ODataSerializerProvider SerializerProvider)
```

Comment:
>value [](http://example.com/codeflow?start=123&length=5)

We are expecting this to be an IODataInstanceAnnotationContainer, so please name it something like instanceAnnotationContainer rather than the generic (and unhelpful) "value".

Reply:
This was kept so because from the calling method it sends an object(got from trygetvalue outs object) . So a cast would be inevitable, may be can move that cast to the calling method